go-controller/pkg/node/port_claim.go

```diff
@@ -108,6 +108,36 @@ func (p *localPortManager) close(desc string, ip string, port int32, protocol ka
 	return fmt.Errorf("error closing socket for svc: %s/%s on port: %v, port was never opened...?", svc.Namespace, svc.Name, port)
 }
 
+func (p *localPortManager) handleService(svc *kapi.Service, handler handler) []error {
+	errors := []error{}
+	if !util.ServiceTypeHasNodePort(svc) && len(svc.Spec.ExternalIPs) == 0 {
+		return errors
```

Comment:
not sure if this is useful or just a personal preferance but I like if there is no error to return nil even if the error is initialized. Feel free to ignore this suggestion